interpreter/interpreter.go

```diff
@@ -1031,21 +1038,22 @@ func resolveValue(v values.Value) (semantic.Node, error) {
 				return
 			}
 			var n semantic.Node
-			n, err = resolveValue(v)
+			n, ok, err := resolveValue(v)
 			if err != nil {
 				return
+			} else if ok {
+				node.Properties = append(node.Properties, &semantic.Property{
+					Key:   &semantic.Identifier{Name: k},
+					Value: n.(semantic.Expression),
+				})
 			}
```

Comment:
Similar case is needed here, if any property cannot be in-lined then the entire object cannot be in-lined.